hdfs-dataset-sink/src/main/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkConfiguration.java

```diff
@@ -0,0 +1,190 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.hadoop.store.StoreException;
+import org.springframework.data.hadoop.store.dataset.DatasetDefinition;
+import org.springframework.data.hadoop.store.dataset.DatasetOperations;
+import org.springframework.data.hadoop.store.dataset.DatasetRepositoryFactory;
+import org.springframework.data.hadoop.store.dataset.DatasetTemplate;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.expression.spel.SpelParseException;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.integration.aggregator.DefaultAggregatingMessageGroupProcessor;
+import org.springframework.integration.aggregator.ExpressionEvaluatingCorrelationStrategy;
+import org.springframework.integration.aggregator.ExpressionEvaluatingReleaseStrategy;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.config.AggregatorFactoryBean;
+import org.springframework.integration.store.MessageGroupStore;
+import org.springframework.integration.store.MessageGroupStoreReaper;
+import org.springframework.integration.store.SimpleMessageStore;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Configuration class for the HDFS DatasetSink.
+ * <p/>
+ * The configuration contains the property 'fsUri' to configure a connection to HDFS as well as the
+ * additional properties for the sink like directory, fileName, codec etc. You can also use the
+ * standard 'spring.hadoop.fsUri' property for specifying the HDFS connection.
+ *
+ * @author Thomas Risberg
+ */
+@Configuration
+@EnableScheduling
+@EnableConfigurationProperties(DatasetSinkProperties.class)
+public class DatasetSinkConfiguration {
+
+	@Autowired
+	private DatasetSinkProperties properties;
+
+	@Bean
+	public MessageChannel toSink() {
+		return new DirectChannel();
+	}
+
+	@Bean
+	@ServiceActivator(inputChannel= Sink.INPUT)
+	FactoryBean<MessageHandler> aggregatorFactoryBean(MessageChannel toSink, MessageGroupStore messageGroupStore) {
+		AggregatorFactoryBean aggregatorFactoryBean = new AggregatorFactoryBean();
+		aggregatorFactoryBean.setCorrelationStrategy(
+				new ExpressionEvaluatingCorrelationStrategy("payload.getClass().name"));
+		String releaseStrategyExpression = "size() == " + properties.getBatchSize();
```

Comment:
Looks like `MessageCountReleaseStrategy` would be better for this kind of aggregator.